libraries/rush-lib/src/api/RushConfigurationProject.ts

```diff
@@ -311,7 +311,20 @@ export class RushConfigurationProject {
     this.unscopedTempProjectName = PackageNameParsers.permissive.getUnscopedName(tempProjectName);
 
     this.decoupledLocalDependencies = new Set<string>();
-    if (projectJson.cyclicDependencyProjects || projectJson.decoupledLocalDependencies) {
+    if (rushConfiguration.experimentsConfiguration.configuration.autoGenerateDecoupledLocalDependencies) {
+      for (const { name, version } of [
+        ...this.packageJsonEditor.dependencyList,
+        ...this.packageJsonEditor.devDependencyList
+      ]) {
+        const dependencySpecifier: DependencySpecifier = new DependencySpecifier(name, version);
+        if (
+          rushConfiguration.rushConfigurationJson.projects.some((project) => project.packageName === name) &&
```

Comment:
This is an `O(n^2 * m)` algorithm cost, where `n` is the # of projects in **rush.json** and `m` is the number of dependencies in **package.json**.  Additionally the `[...` notation and `projects.some(` notation will allocate lots of short-lived GC objects.  Since this code runs for many Rush commands, I think it could be a performance concern.